src/modules/Dashboard/features/Participant/Assignments/ByParticipant/ByParticipant.tsx

```diff
@@ -1,5 +1,181 @@
+
+    fetchActivities({ appletId, subjectId: respondentSubject.id });
+  }, [appletId, fetchActivities, respondentSubject]);
+
+  const handleRefetchAll = useCallback(() => {
+    handleRefetchActivities();
+
+    // Refresh target subject data for any expanded views
+    Object.entries(expandedViewsData).forEach(([id, viewData]) => {
+      if (viewData) handleRefetchExpandedView(id);
+    });
+  }, [handleRefetchActivities, expandedViewsData, handleRefetchExpandedView]);
+
+  const {
+    getActionsMenu,
+    onClickAssign,
+    onClickNavigateToData,
+    isLoading: isLoadingHook,
+    modals,
+  } = useAssignmentsTab({
+    appletId,
+    respondentSubject,
+    handleRefetch: handleRefetchAll,
+    dataTestId,
+  });
+
+  const handleClickNavigateToData = (
+    activityOrFlow: ParticipantActivityOrFlow,
+    targetSubject: RespondentDetails,
+  ) => {
+    if (!respondentSubject) return;
+
+    onClickNavigateToData(activityOrFlow, targetSubject.id);
+  };
+
+  const handleClickToggleExpandedView = useCallback(
+    (isExpanded: boolean, activityOrFlowId: string) => {
+      if (!respondentSubject) return;
+
+      if (isExpanded) {
+        handleRefetchExpandedView(activityOrFlowId);
+      } else {
+        // If expanded view is closed, remove data to free up memory and minimize refetches
+        // (after delay to account for transition)
+        setTimeout(() => {
+          setExpandedViewsData((prev) => ({ ...prev, [activityOrFlowId]: undefined }));
+        }, 300);
+      }
+    },
+    [handleRefetchExpandedView, respondentSubject],
+  );
+
+  useEffect(() => {
+    handleRefetchActivities();
+  }, [handleRefetchActivities]);
+
+  const isLoading = isLoadingRespondentSubject || isLoadingActivities || isLoadingHook;
+
+  return (
+    <AssignmentsTab>
+      {isLoading && <Spinner />}
+
+      {!isLoading && !activities.length && (
+        <EmptyState onClickAssign={onClickAssign} isLimitedAccount={!respondentSubject?.userId} />
+      )}
+
+      {!!activities.length && respondentSubject && (
+        <ActivitiesList
+          title={t('participantDetails.activitiesAndFlows')}
+          count={fetchedActivities?.data.count ?? 0}
+        >
+          {activities.map((activity, index) => (
+            <ActivityListItem
+              key={activity.id}
+              activityOrFlow={activity}
+              onClickToggleExpandedView={(isExpanded) =>
+                handleClickToggleExpandedView(isExpanded, activity.id)
+              }
+              expandedView={
+                <ExpandedView
+                  activityOrFlow={activity}
+                  targetSubjects={expandedViewsData[activity.id]}
+                  getActionsMenu={getActionsMenu}
+                  onClickViewData={handleClickNavigateToData}
+                  data-test-id={`${dataTestId}-${index}`}
+                />
+              }
+              isLoadingExpandedView={!!expandedViewsLoading[activity.id]}
```

Comment:
This value is already a boolean
```suggestion
              isLoadingExpandedView={expandedViewsLoading[activity.id]}
```